### PR TITLE
ScyllaDB Swagger client

### PR DESCRIPTION
**Pull request by zimnx**
Swagger client provides a base for building HTTP clients. Autogenerated code from this repo, can be consumed by wrapper clients providing better UX.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,105 @@
+all: update
```
Comment (tnozicka) [inline Makefile:1]: I am a bit skeptical to changing the default build to update that overwrites files. To add to it, I think a build of stuff that was generated is useful as well and should likely be exercised in the CI to make sure the generated code compiles.

File: Makefile [hunk 2]
```diff
@@ -19,6 +19,11 @@ go_packages_dirs :=$(shell $(GO) list -f '{{ .Dir }}' $(GO_PACKAGES) || echo 'no
 
 GO_VERSION :=$(shell $(GO) version | sed -E -e 's/.*go([0-9]+.[0-9]+.[0-9]+).*/\1/')
 
+SWAGGER ?= $(GO) run ./vendor/github.com/go-swagger/go-swagger/cmd/swagger
```
[the remark this answers is not in the document]
Comment (zimnx) [reply Makefile]: done

File: Makefile [hunk 3]
```diff
@@ -45,6 +50,38 @@ ifneq "$(GO_REQUIRED_MIN_VERSION)" ""
...
+# $1 - application name
+# $2 - spec file to use
+# $4 - template directory
+# $3 - target directory
+define generate-swagger-client
+	$(SWAGGER) generate client --name='$(1)' --spec='$(2)' --template-dir='$(3)' --target='$(4)'
+endef
+
+update-scylladbclient:
+	$(call generate-swagger-client,'scylladb_v1','$(SCYLLADB_SWAGGER_PATH)/scylladb_v1.json','$(SCYLLADB_SWAGGER_PATH)/templates','$(SCYLLADB_SWAGGER_GEN_PATH)/v1')
```
[the remark this answers is not in the document]
Comment (zimnx) [reply Makefile]: done

File: README.md [hunk 4]
```diff
@@ -2,5 +2,43 @@
...
+
+```go
+package main
+
+import (
+    "fmt"
+
+    api "github.com/go-openapi/runtime/client"
+    "github.com/go-openapi/strfmt"
+    scyllaclient "github.com/scylladb/scylladb-swagger-go-client/scylladb/gen/v1/client"
```
[the remark this answers is not in the document]
Comment (zimnx) [reply README.md]: done

File: README.md [hunk 5]
```diff
@@ -2,5 +2,43 @@
...
+    scyllaoperations "github.com/scylladb/scylladb-swagger-go-client/scylladb/gen/v1/client/operations"
+)
+
+func main() {
+    // Initialize the ScyllaDB client
+    runtime := api.New("127.0.0.1:10000", scyllaclient.DefaultBasePath, scyllaclient.DefaultSchemes)
+    scyllaClient := scyllaclient.New(runtime, strfmt.Default)
+
+    // Get all hosts
+    resp, _ := scyllaClient.Operations.StorageServiceHostIDGet(&scyllaoperations.StorageServiceHostIDGetParams{})
```
[the remark this answers is not in the document]
Comment (zimnx) [reply README.md]: done

File: .github/workflows/go.yaml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        path: ${{ env.git_repo_path }}
+    - name: Setup go
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.go_version }}
+    - name: Verify
+      run: make verify --warn-undefined-variables
```
[the remark this answers is not in the document]
Comment (zimnx) [reply .github/workflows/go.yaml:54]: done

File: Makefile (hunk 1 above)
Comment (zimnx) [reply Makefile]: done

**Review by tnozicka — APPROVED** [review]
/approve
one last nit, thanks

File: .github/workflows/go.yaml [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+      with:
+        path: ${{ env.git_repo_path }}
+    - name: Setup go
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ env.go_version }}
+    - name: Verify dependencies
+      run: make verify-deps --warn-undefined-variables
+  
+  build-and-test:
```
[the remark this answers is not in the document]
Comment (zimnx) [reply .github/workflows/go.yaml]: done

**Pull request merged**